library/std/src/fs/tests.rs

```diff
@@ -901,12 +907,26 @@ fn symlink_noexist() {
 #[test]
 fn read_link() {
     if cfg!(windows) {
+        let environment_variable = "CI";
         // directory symlink
         assert_eq!(check!(fs::read_link(r"C:\Users\All Users")), Path::new(r"C:\ProgramData"));
         // junction
         assert_eq!(check!(fs::read_link(r"C:\Users\Default User")), Path::new(r"C:\Users\Default"));
         // junction with special permissions
-        assert_eq!(check!(fs::read_link(r"C:\Documents and Settings\")), Path::new(r"C:\Users"));
+        match env::var(environment_variable) {
+            Ok(_var) => assert_eq!(
+                check!(fs::read_link(r"C:\Documents and settings\")),
+                Path::new(r"C:\Users")
+            ),
+            Err(_e) => {
+                if Path::new(r"C:\Documents and settings\").exists() {
+                    assert_eq!(
+                        check!(fs::read_link(r"C:\Documents and settings\")),
+                        Path::new(r"C:\Users")
+                    )
+                }
+            }
+        }
```

Comment:
Probably simpler to do something like this: 
```rust
if Path::new(r"C:\Documents and settings\").exists() || env::var_os("CI").is_some() {
    assert_eq!(
        check!(fs::read_link(r"C:\Documents and settings\")), Path::new(r"C:\Users"));
}
```
Also some comment explaining that C:\Documents and settings\ sometimes does not exist on localized Windows would be good IMHO.